synapse/replication/tcp/client.py

```diff
@@ -154,7 +154,8 @@ async def on_rdata(
                 max_token = self.store.get_room_max_stream_ordering()
```

Comment:
This is calculating the same thing it looks like, can we avoid doing it twice?

Reply:
Yes, but that line is a) cheap and b) about to die in my next PR :)

Reply:
For some reason I keep thinking that this causes database access. 🤦

Reply:
That is not an unreasonable assumption given its in `self.store`....

Reply:
Anyway, I think it looks fine!